benchmarks/src/main/kotlin/chat/BenchmarkConfiguration.kt

```diff
@@ -0,0 +1,192 @@
+package chat
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.scheduling.ExperimentalCoroutineDispatcher
+import org.apache.commons.math3.stat.descriptive.moment.StandardDeviation
+import java.util.concurrent.ForkJoinPool
+
+// Benchmarks configuration to monitor
+
+/**
+ * Underlying thread pool size for coroutines.
+ */
+val THREADS = listOf(4, 8, 16, 1)
+/**
+ * Chat users count.
+ */
+val USER_COUNT = listOf(10000)
+/**
+ * Maximum percentage of all chat users a user can be friends with.
+ */
+val MAX_FRIENDS_PERCENTAGE = listOf(0.2)
+/**
+ * Coroutines channels type.
+ */
+val CHANNEL_TYPES = listOf(ChannelType.UNLIMITED, ChannelType.RENDEZVOUS, ChannelType.BUFFERED_1, ChannelType.BUFFERED_16, ChannelType.BUFFERED_256)
+/**
+ * The average amount work that will be executed on CPU.
+ */
+val AVERAGE_WORK = listOf(80, 40)
+/**
+ * Implementations of users that contain different ways to find a user to send a message.
+ */
+val BENCHMARK_MODE = listOf(BenchmarkModes.USER_WITH_FRIENDS, BenchmarkModes.USER_WITHOUT_FRIENDS)
+/**
+ * Coroutines dispatcher types
+ */
+val DISPATCHER_TYPES = listOf(DispatcherTypes.EXPERIMENTAL, DispatcherTypes.FORK_JOIN)
+
+// Benchmark configuration that are not shown in the output file
+
+/**
+ * Execution time of each benchmark.
+ */
+const val BENCHMARK_TIME_MS = 2000L
+/**
+ * Warm up iterations count
+ */
+const val WARM_UP_ITERATIONS = 5
+/**
+ * Benchmark iterations count
+ */
+const val BENCHMARK_ITERATIONS = 5
+/**
+ * CSV file containing the configurations and final metrics of the executed benchmarks
+ */
+const val BENCHMARK_OUTPUT_FILE = "results.csv"
+/**
+ * Folder containing all benchmark output files
+ */
+const val BENCHMARK_OUTPUT_FOLDER = "src/main/kotlin/chat/benchmarkresults/"
+/**
+ * The header for the output CSV file. It should be coordinated with [chat.BenchmarkConfiguration.toCSV] method.
+ */
+const val BENCHMARK_CONFIGURATION_FIELDS = "threads,userCount,maxFriendsPercentage,channelType,averageWork,benchmarkMode,dispatcherType,avgSentMessages,stdSentMessages,avgReceivedMessages,stdReceivedMessages"
```

Comment:
It is better to create a special class for benchmark results and have a method for writing them as a CSV file